lib/src/workspace.rs

```diff
@@ -119,7 +125,17 @@ pub struct Workspace {
 fn create_jj_dir(workspace_root: &Path) -> Result<PathBuf, WorkspaceInitError> {
     let jj_dir = workspace_root.join(".jj");
     match std::fs::create_dir(&jj_dir).context(&jj_dir) {
-        Ok(()) => Ok(jj_dir),
+        Ok(()) => {
+            #[cfg(windows)]
+            #[allow(unsafe_code)]
+            {
+                let jj_dir = CString::new(jj_dir.to_str().unwrap()).unwrap();
+                unsafe {
+                    SetFileAttributesA(jj_dir.as_ptr(), FILE_ATTRIBUTE_HIDDEN);
```

Comment:
Perhaps, we should use the "W" API by using the underlying `OsString` data. I don't know if there's a safer/convenient wrapper, though. We'll also need to check the error code somehow.